framework/src/utils/SlepcSupport.C

```diff
@@ -992,38 +1004,40 @@ attachCallbacksToMat(EigenProblem & eigen_problem, Mat mat, bool eigen)
   // Attach the Jacobian computation function. If \p mat is the "eigen" matrix corresponding to B,
   // then attach our JacobianB computation routine, else the matrix corresponds to A, and we attach
   // the JacobianA computation routine
-  PetscObjectComposeFunction((PetscObject)mat,
-                             "formJacobian",
-                             eigen ? Moose::SlepcSupport::mooseSlepcEigenFormJacobianB
-                                   : Moose::SlepcSupport::mooseSlepcEigenFormJacobianA);
+  LIBMESH_CHKERR(
```

Comment:
This is an interesting idiom for it.  Any reason for preferring this here vs a named `ierr` elsewhere?